poppler: 21.02.0 -> 21.05.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/metapackage/poppler-glib/versions.


meta.description for poppler is: "A PDF rendering library"

meta.homepage for poppler is: "https://poppler.freedesktop.org/"

meta.changelog for poppler is: ""


###### Updates performed
- Version update

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 21.03.0 with grep in /nix/store/75hr95dnbp0ckn8vm1r9yblvb29nq751-poppler-glib-21.03.0
- found 21.03.0 in filename of file in /nix/store/75hr95dnbp0ckn8vm1r9yblvb29nq751-poppler-glib-21.03.0
- directory tree listing: https://gist.github.com/c170b7012de5716dbe4d23943a9ffa14

- du listing: https://gist.github.com/39804706275b58adca4e4f2d683bc73c

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
1844 total rebuild path(s)

604 package rebuild(s)

604 x86_64-linux rebuild(s)
540 i686-linux rebuild(s)
171 x86_64-darwin rebuild(s)
529 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
R
adapta-gtk-theme
aesop
anystyle-cli
apostrophe
apvlv
arc-theme
arx-libertatis
asciidoc-full
asciidoc-full-with-plugins
asymptote
auctex
autofs5
beamerpresenter
bibata-cursors
bibata-cursors-translucent
bibata-extra-cursors
bookworm
btrbk
calibre
calligra
capitaine-cursors
ccache
ccacheStdenv
ccacheWrapper
cddlib
chrome-gnome-shell
cinnamon.cinnamon-common
cinnamon.cinnamon-control-center
cinnamon.cinnamon-gsettings-overrides
cinnamon.cinnamon-screensaver
cinnamon.cinnamon-session
cinnamon.cinnamon-settings-daemon
cinnamon.iso-flags-png-320x420
cinnamon.iso-flags-svg
claws-mail
claws-mail-gtk3
clevis
cloudcompare
colord
colord-gtk
cups-filters
darktable
dblatex
dblatexFull
denemo
diff-pdf
diffpdf
digikam
disorderfs
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/75hr95dnbp0ckn8vm1r9yblvb29nq751-poppler-glib-21.03.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A poppler https://github.com/r-ryantm/nixpkgs/archive/d2d1b6561b90f1ac9cb8101abc7514081d490b87.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/75hr95dnbp0ckn8vm1r9yblvb29nq751-poppler-glib-21.03.0
ls -la /nix/store/75hr95dnbp0ckn8vm1r9yblvb29nq751-poppler-glib-21.03.0/bin
```

---

</details>
<br/>



### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings

cc @ttuegel @jtojnar @worldofpeace for testing.

jtojnar reviewed (approved):
Changes look okay: https://lists.freedesktop.org/archives/poppler/2021-March/014701.html

jtojnar commented:
@GrahamcOfBorg build inkscape cups-filters texlive scribusUnstable

jtojnar commented:
It looks like scribusUnstable failed to build. If that is more than just a spurious failure, someone will need to patch it.

jtojnar committed:
poppler: clean up

- reorder expression to more traditional order
- make inputs one per line
- correct license
- add passthru.tests

jtojnar committed:
poppler: 21.02.0 -> 21.05.0

Changes:
- https://lists.freedesktop.org/archives/poppler/2021-March/014701.html
- https://lists.freedesktop.org/archives/poppler/2021-April/014720.html
- https://lists.freedesktop.org/archives/poppler/2021-May/014743.html

jtojnar commented:
These three patches might fix `scribusUnstable` but I cannot test them now:

- https://github.com/scribusproject/scribus/commit/7ce0ac16fd42d61ef9082b27822c7d9d79c7fef7.patch
- https://github.com/scribusproject/scribus/commit/6b9ff916959bcb941866f0bd86da639a421337f8.patch
- https://github.com/scribusproject/scribus/commit/c62844064cd6d85802d21e188b0f479463e22095.patch

purcell commented:
Do we need to fix `scribusUnstable` for this to move ahead? This already fixes the broken state of the rather popular `gdal` and `postgis` libraries: even though that's on Darwin, perhaps that's a higher priority, assuming there's a "regular" scribus that still works?

jtojnar commented:
The regular `scribus` is ancient and effectively unmaintained, and what we call “unstable” is [considered stable by upstream](https://www.scribus.net/scribus-1-5-7-released/). And IMO infinitely more useful than the stale version, just like with NixOS.

jtojnar commented:
Hmm, it looks like `scribusUnstable` actually builds so it must have been a spurious CI failure.

purcell commented:
Oh yay, thanks @jtolnar.
